service/storage_proxy.cc

```diff
@@ -411,15 +411,15 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
         --_stats.writes;
         if (_cl_achieved) {
             if (_throttled) {
-                _ready.set_value();
+                _ready.set_value(bo::success());
             } else {
                 _stats.background_writes--;
                 _proxy->_global_stats.background_write_bytes -= _mutation_holder->size();
                 _proxy->unthrottle();
             }
         } else {
             if (_error == error::TIMEOUT) {
-                _ready.set_exception(mutation_write_timeout_exception(get_schema()->ks_name(), get_schema()->cf_name(), _cl, _cl_acks, _total_block_for, _type));
+                _ready.set_value(coordinator_failure(mutation_write_timeout_exception(get_schema()->ks_name(), get_schema()->cf_name(), _cl, _cl_acks, _total_block_for, _type)));
```

Comment:
Do we need to go through the `coordinator_failure`. Is there c++ magic that can convert anything that inherits from std::exception into an error result?

Reply:
Oh, it turns out there are implicit constructors from the value type and error type. `_ready.set_value(mutation_write_timeout_exception(...))` just works. I'll remove the `coordinator_failure` function altogether.

I previously thought that `bo::failure()` was mandatory for constructing a result and tried doing `_ready.set_value(bo::failure(mutation_write_timeout_exception(...)))`, which didn't work so I wrote the helper function.